lib/model.js

```diff
@@ -1641,6 +1641,10 @@ Model.prototype.findOrCreate = function(options) {
     );
   }
 
+  if (this.sequelize.constructor.cls) {
+    options.transaction = this.sequelize.constructor.cls.get('transaction');
```

Comment:
Please follow the same style as `query`. https://github.com/gutobortolozzo/sequelize/blob/clsFindOrCreate/lib/sequelize.js#L689
Use the Sequelize instance rather than access `constructor`